symforce/slam/imu_preintegration/imu_factor.h

```diff
@@ -0,0 +1,191 @@
+/* ----------------------------------------------------------------------------
+ * SymForce - Copyright 2022, Skydio, Inc.
+ * This source code is under the Apache 2.0 license found in the LICENSE file.
+ * ---------------------------------------------------------------------------- */
+
+#pragma once
+
+#include <vector>
+
+#include <Eigen/Core>
+
+#include <sym/pose3.h>
+#include <symforce/opt/factor.h>
+#include <symforce/opt/key.h>
+
+#include "./imu_preintegrator.h"
+#include "./preintegrated_imu_measurements.h"
+
+namespace sym {
+
+/**
+ * A factor for using on-manifold IMU preintegration in a SymForce optimization problem.
+ *
+ * Assumes IMU bias is constant over the preintegration window. Does not recompute the
```

Comment:
It would be good to include a description of we mean by "on-manifold" here

Reply:
That's reasonable. Though, I've never actually seen a definition of tangent space pre-integration (I've neglected to read through the gtsam implementation details, and that's the only source I can think of as my googling prowess is not up to par). So this is what I've put and hopefully it is correct:
```
 * A factor for using on-manifold IMU preintegration in a SymForce optimization problem. By
 * on-manifold, it is meant that the angular velocity measurements are composed as rotations
 * rather than tangent-space approximations.
```

Reply:
I believe this is correct, I might say something like "By on-manifold, we mean that the integration is performed (and integrated states are represented) on the manifold, as opposed to in the tangent space of the initial state"

Reply:
A short discussion of the benefits / differences of the on0manifold vs tangent representations would be great, if possible.